geonode/security/models.py

```diff
@@ -388,3 +388,46 @@ def set_workflow_perms(self, approved=False, published=False):
             if settings.OGC_SERVER['default'].get("GEOFENCE_SECURITY_ENABLED", False):
                 if self.polymorphic_ctype.name == 'layer':
                     sync_geofence_with_guardian(self.layer, VIEW_PERMISSIONS)
+
+    def get_user_perms(self, user):
+        """
+        Returns a list of permissions a user has on a given resource
+        """
+        resource = self.get_self_resource()
+        user_resource_perms = get_users_with_perms(resource).get(user, [])
```

Comment:
@afabiani we should avoid `get_users_with_perms() ` because it iterates over all users each time. We should have the option to only ask for a specific user and filter the queryset, instead of iterating.